### PR TITLE
Allow multiple changelog files

### LINKED ISSUE #59: Multiple Changelog files?
I have enjoyed using `releases` on several projects and am looking to use it on several more.

I am wondering if it is possible to process multiple "changelog" files?

The particular situation I have is that I have several small projects around a common theme. I would like to maintain a changelog per project, but would like to maintain only one documentation site for the lot of them. Suggestions on how to do this?

### PR DESCRIPTION
**Pull request by MinchinWeb**
Fixes #59

### CODE REVIEW

File: releases/__init__.py [hunk 1]
```diff
@@ -629,6 +629,12 @@ def setup(app):
         app.add_config_value(
             name='releases_{0}'.format(key), default=default, rebuild='html'
         )
+    # if a string is given for `document_name`, convert it to a list
+    # done to maintain backwards compatibility
+    if isinstance(app.config.releases_document_name, str):
```
Comment (bitprophet) [inline releases/__init__.py]: Probably best to `s/str/six.string_types` here in case anyone's on Python 2 + using Unicode string objects.

**Comment by bitprophet** [comment]
Curious why you're using "single-item tuple" style lists here? (i.e. `['foo', ]` instead of `['foo']`) - it's not standard style. Dunno if a linter would complain about it but I would if I was a linter :D

**Comment by bitprophet** [comment]
Also, if you could add an integration test for this (there's existing ones for "use a different filename" for example) that'd be rad!

Besides that, the above nitpick, and the line comment I left, looks good - thanks!

**Comment by MinchinWeb** [comment]
I'm not sure where I picked up the "simple-item tuple"-style lists. Probably something wasn't turning into a list like I was wanting it to, so I made it a little more explicit. In any case, the code here as been updated to pass the linter :D

I added an integration test. That turned out to be at least twice as hard as adding the functionality in the first place. The biggest stumbling block proved to be that I couldn't figure out how to pass a python list via the command-line, so I had to set up the integration tests to accept an alternate configuration file and not to pass the `-D` option. But everything is working now!

**Comment by bitprophet** [comment]
Huh, weird - not sure offhand why that was necessary, typically just "another directory w/ docs and conf.py" is enough! I might poke when I merge, but otherwise, glad you got it working :)

File: releases/__init__.py [hunk 2]
```diff
@@ -598,8 +598,8 @@ def unknown_visit(self, node):
 
 def generate_changelog(app, doctree):
     # Don't scan/mutate documents that don't match the configured document name
-    # (which by default is changelog.rst).
-    if app.env.docname != app.config.releases_document_name:
+    # (which by default is ['changelog.rst', ]).
+    if app.env.docname not in app.config.releases_document_name:
```
Comment (MinchinWeb) [inline releases/__init__.py]: If you feed a list in as `app.config.release_document_name`, re breaks with the error `TypeError: unhashable type: 'list'`.

Alternately, if you feed `*/changelog.rst` in as `app.config.release_document_name`, re breaks with the error `sre_comstraints.error: nothing to repeat at position 0`.

I feel like I'm missing something here in what you're proposing...

Comment (jta) [inline releases/__init__.py]: Sorry, it was off the cuff remark, I didn't check. The correct syntax based on that error message would be `.*/changelog.rst` rather than the glob syntax I used.

My point was that defining a list of files is cumbersome if you have a large project with an unknown set of changelogs. Regexes tend to be much more practical for such cases.

Comment (MinchinWeb) [inline releases/__init__.py]: I keep thinking on this. I like the idea of being able to simply specify a range of documents, but regex can be a pain to debug at the best of times.

Since this is passed from the Sphinx configuration file, which can execute arbitrary Python, would it make sense to have the user generate their own list somehow? Maybe add an example to the documentation.

Another option would be to pass it through a glob-er (is that even a word?) rather than a regex. Maybe this is done internally, maybe it is pushed to the config file. For example:

```
from pathlib import Path
p = Path('.')
list(p.glob(**/changelog.rst))
```

Comment (bitprophet) [inline releases/__init__.py]: Offhand I do prefer to leverage the fact that `conf.py` is a `.py`, so +1 to "just ask user to supply a list". Or the user could use `glob.glob` from stdlib, or third party modules like `pathlib` as noted.

tl;dr when a use case is uncommon-to-rare, enabling it via "user is enabled to write their own tiny bit of Python" is preferable over "we write our own 'blessed' tiny-bit-of-Python and expose as a func/feature".

So as long as we allow a list value in the setting, the user is free to generate it however they want, and we're neither assisting nor limiting them.

**Comment by bitprophet** [comment]
Merging this now and oh no, the integration test is actually a false positive 😆 it only runs the 1st ('A') changelog; the 2nd is not actually built or run. Wish I'd noticed this before spending 15 minutes tearing my hair out over why a sanity 'break' of the B changelog was not resulting in a test failure...

**Comment by bitprophet** [comment]
(FTR I am working on that and any other necessary fixes on my end.)

**Comment by bitprophet** [comment]
Solved that, made tweaks, applied to my real world use case, seems to work! Merged, will release as 1.5.0 soon.
